### PR TITLE
Typos, errors and weirdness fixing

### PR DESCRIPTION
**Pull request by Bertie690**
probably gonna have some merge conflicts with other branches but whatever

### CODE REVIEW

File: cs/traits.go [hunk 1]
```diff
@@ -499,7 +502,7 @@ func urSpec() LRTSpec {
 	return LRTSpec{
 		// UR gives us 45%/90% of scrapped minerals, versus 33%/80% for races without UR
 		ScrapMineralOffset:           .45 - (1.0 / 3),
-		ScrapMineralOffsetStarbase:   .9 - (1.0 / 3),
+		ScrapMineralOffsetStarbase:   .9 - (10 / 8),
```
Comment (sirgwain) [inline cs/traits.go]: This will make the result `-.35`

File: cs/techstore.go [hunk 2]
```diff
@@ -203,250 +205,296 @@ func (store *TechStore) GetHullComponentsByCategory(category TechCategory) []Tec
...
 }
 
-// get the best terraform for a player
+// get the player's best terraform
 func (store *TechStore) GetBestTerraform(player *Player, terraformHabType TerraformHabType) (bestTech *TechTerraform) {
 	for i := range store.Terraforms {
 		tech := &store.Terraforms[i]
 		if tech.HabType == terraformHabType && player.HasTech(&tech.Tech) {
-			bestTech = tech
+			if tech.Tech.Ranking > bestTech.Tech.Ranking {
```
Comment (sirgwain) [inline cs/techstore.go]: This needs a nil check

**Review by sirgwain — CHANGES_REQUESTED** [review]

File: cs/universegenerator.go [hunk 3]
```diff
@@ -297,83 +297,90 @@ func (ug *universeGenerator) generatePlayerHomeworlds(area Vector) error {
...
+			} else {
+				surface = extraWorldSurfaceMinerals
 			}
 
 			// make a new starter world
-			playerPlanet.initStartingWorld(player, &ug.Rules, startingPlanet, homeworldMinConc, surfaceMinerals)
+			playerPlanet.initStartingWorld(player, &ug.Rules, startingPlanet, homeworldMinConc, surface)
+			if !startingPlanet.Homeworld && !ug.MaxMinerals {
+				// starting planets start with different mincons
+				playerPlanet.MineralConcentration = randomizeMinerals(rules, playerPlanet.Hab.Rad)
```
Comment (sirgwain) [inline cs/universegenerator.go:368]: we can't randomize this, all extra worlds should be the same. If two players player IT, one shouldn't get an extra world with bum minerals.

Comment (Bertie690) [reply cs/universegenerator.go:368]: Well, I just did it based on what OG stars! did, since they also randomized extra world mincons & hab.

Comment (sirgwain) [reply cs/universegenerator.go:368]: ah, so they do. nm!

**Review by sirgwain — APPROVED** [review]

**Pull request merged**